packages/flutter_test/test/controller_test.dart

```diff
@@ -0,0 +1,277 @@
+        <Offset>[
+          const Offset(10.0, 10.0),
+          const Offset(0.0, 5.0),
+          const Offset(0.0, 5.0),
+        ],
+
+        //// [VARYING TOUCH SLOP] ////
+        <Offset>[
+          const Offset(12.0, 5.0),
+          const Offset(0.0, 5.0),
+          const Offset(0.0, 5.0),
+        ],
+        <Offset>[
+          const Offset(12.0, 5.0),
+          const Offset(20.0, 5.0),
+          const Offset(12.0, 3.0),
+          const Offset(8.0, 2.0),
+        ],
+        <Offset>[
+          const Offset(12.0, 5.0),
+          const Offset(5.0, 20.0),
+          const Offset(1.25, 5.0),
+          const Offset(3.75, 15.0),
+        ],
+        <Offset>[
+          const Offset(5.0, 12.0),
+          const Offset(5.0, 20.0),
+          const Offset(3.0, 12.0),
+          const Offset(2.0, 8.0),
+        ],
+        <Offset>[
+          const Offset(5.0, 12.0),
+          const Offset(20.0, 5.0),
+          const Offset(5.0, 1.25),
+          const Offset(15.0, 3.75),
+        ],
+        <Offset>[
+          const Offset(18.0, 18.0),
+          const Offset(0.0, 150.0),
+          const Offset(0.0, 18.0),
+          const Offset(0.0, 132.0),
+        ],
+        <Offset>[
+          const Offset(18.0, 18.0),
+          const Offset(0.0, -150.0),
+          const Offset(0.0, -18.0),
+          const Offset(0.0, -132.0),
+        ],
+        <Offset>[
+          const Offset(18.0, 18.0),
+          const Offset(-150.0, 0.0),
+          const Offset(-18.0, 0.0),
+          const Offset(-132.0, 0.0),
+        ],
+        <Offset>[
+          const Offset(0.0, 0.0),
+          const Offset(-150.0, 0.0),
+          const Offset(-150.0, 0.0),
+        ],
+        <Offset>[
+          const Offset(18.0, 18.0),
+          const Offset(-32.0, 0.0),
+          const Offset(-18.0, 0.0),
+          const Offset(-14.0, 0.0),
+        ],
+      ];
+
+      await tester.pumpWidget(
+        const Directionality(
+          textDirection: TextDirection.ltr,
+          child: Text('test'),
+        ),
+      );
+
+      final WidgetControllerSpy spyController = WidgetControllerSpy(tester.binding);
+
+      for (int resultI = 0; resultI < offsetResults.length; resultI += 1) {
+        final List<Offset> testResult = offsetResults[resultI];
+        await spyController.drag(
+          find.text('test'),
+          testResult[1],
+          touchSlopX: testResult[0].dx,
+          touchSlopY: testResult[0].dy,
+        );
+        final List<Offset> dragOffsets = spyController.testGestureSpy.offsets;
+        expect(
+          offsetResults[resultI].length - 2,
+          dragOffsets.length,
+
+          reason:
+            'There is a difference in the number of expected and actual split offsets for the drag with:\n'
+            'Touch Slop: ' + testResult[0].toString() + '\n'
+            'Delta:      ' + testResult[1].toString() + '\n',
+        );
+        for (int valueI = 2; valueI < offsetResults[resultI].length; valueI += 1) {
```

Comment:
what's the "I" in "resultI" and "valueI"?

Reply:
Index